apps/web/lib/rewardful/import-campaign.ts

```diff
@@ -25,34 +26,51 @@ export async function importCampaign({ programId }: { programId: string }) {
     reward_type,
   } = campaign;
 
-  const reward = await prisma.reward.create({
-    data: {
-      id: createId({ prefix: "rw_" }),
-      programId,
-      event: "sale",
-      maxDuration: max_commission_period_months,
-      type: reward_type === "amount" ? "flat" : "percentage",
-      amount:
-        reward_type === "amount" ? commission_amount_cents : commission_percent,
-    },
+  const newReward = {
+    programId,
+    event: EventType.sale,
+    maxDuration: max_commission_period_months,
+    type:
+      reward_type === "amount"
+        ? CommissionType.flat
+        : CommissionType.percentage,
+    amount:
+      reward_type === "amount" ? commission_amount_cents : commission_percent,
+  };
+
+  const rewardFound = await prisma.reward.findFirst({
+    where: newReward,
   });
 
-  if (!defaultRewardId) {
-    await prisma.program.update({
-      where: {
-        id: programId,
-      },
+  let rewardId: string | null = null;
+
+  if (!rewardFound) {
+    const reward = await prisma.reward.create({
       data: {
-        defaultRewardId: reward.id,
+        id: createId({ prefix: "rw_" }),
+        ...newReward,
       },
     });
+
+    if (!defaultRewardId) {
+      await prisma.program.update({
+        where: {
+          id: programId,
+        },
+        data: {
+          defaultRewardId: reward.id,
+        },
+      });
+    }
+
+    rewardId = reward.id;
   }
 
   return await rewardfulImporter.queue({
     programId,
     // we will only need to assign rewardId to affiliates
     // if it's not the defaultRewardId of the program (there's already a defaultRewardId)
-    ...(defaultRewardId && { rewardId: reward.id }),
+    ...(defaultRewardId && rewardId && { rewardId }),
```

Comment:
Even if the reward exists (`rewardFound`), don't we still need to pass the rewardId to the users if it's not the `defaultRewardId` so it can be assigned to the partners?

Also, how would program-wide click/lead rewards work?